### PR TITLE
Add VolumeIndex back/forwards compatible to info files + restore flush version on shard bootstrap

### PR DESCRIPTION
**Pull request by richardartoul**
**What this PR does / why we need it**:
This P.R add the `VolumeIndex` into the Info file of all the fileset files in a backwards and forwards compatible way. It also ensures that the volume index is written out by the writers and return by calls to `ReadInfoFile`.

Finally, it updates the shard bootstrap logic to restore the Cold flush version on bootstrap as well as the flush state.

### CODE REVIEW

File: src/dbnode/storage/shard.go [hunk 1]
```diff
@@ -1872,6 +1872,11 @@ func (s *dbShard) Bootstrap(
 		}
 
 		s.markWarmFlushStateSuccess(at)
+		// Cold version needs to get bootstrapped so that the 1:1 relationship between volume number
+		// and cold version is maintained and the volume numbers / flush versions remaining monotonically
+		// increasing.
+		// TODO(rartoul): Add a test for this before merging.
```
Comment (robskillington) [inline src/dbnode/storage/shard.go]: super nit: Want to use the TODO_COLD_FLUSH... flag?

Comment (robskillington) [reply src/dbnode/storage/shard.go]: Oh before merging, never mind.

Comment (richardartoul) [reply src/dbnode/storage/shard.go]: added the coverage

**Comment by justinjc** [comment]
LGTM. Pushed this together with #1762 on a test cluster and everything looked good in terms of backwards- and forwards-compatibility (from >=0.11.0).

**Pull request merged**